rskj-core/src/main/java/co/rsk/peg/Federation.java

```diff
@@ -49,7 +49,7 @@ public class Federation {
     protected Script p2shScript;
     protected Address address;
 
-    public Federation(List<FederationMember> members, Instant creationTime, long creationBlockNumber,  NetworkParameters btcParams) {
+    public Federation(List<FederationMember> members, Instant creationTime, long creationBlockNumber, NetworkParameters btcParams) {
```

Comment:
We should have a look at the method `getNumberOfSignaturesRequired`. I don't think that implementation is correct. We should get the required signers from the OP_M contained in the redeem script, that's the source of truth.

The RedeemScriptParser in bitcoinj-thin has an `getM` method, but I think that implementation might be flawed also. Let's have a look at that

Reply:
Yes, i noticed that when doing changes in bitcoinj-thin. it won't be that easy since the threshold is represented as an OP_PUSHNUM_M, pretty sure it is not an opcode.